### PR TITLE
9289 formalize reflexive

### PR DESCRIPTION
**Pull request by rmulvey**

### CODE REVIEW

**Comment by rmulvey** [comment]
Note that there is a change in here to ParserAllActivityModifier.java. This is a small, pretty insignificant change I made when review a recent change by Levi. I forgot to push this to master , and thus it is included here.

File: src/org.xtuml.bp.core/models/org.xtuml.bp.core/ooaofooa/Association/Simple Association/Simple Association.xtuml [hunk 1]
```diff
@@ -370,14 +370,17 @@ INSERT INTO O_TFR
...
-	'select any part related by self->R_PART[R207]
+	'select many parts related by self->R_PART[R207]
     where selected.Obj_ID == param.id;
-if ( not_empty part )
-  select any oida related by part->R_RTO[R204]->R_OIR[R203]->O_OBJ[R201]->O_ID[R104]->O_OIDA[R105];
-  // participant has to have an id with attributes and cardinality of one (value zero)
-  return not_empty oida and part.Mult == 0;
-end if;
-return false;',
+canParticipate = false;
```
Comment (cortlandstarrett) [inline src/org.xtuml.bp.core/models/org.xtuml.bp.core/ooaofooa/Association/Simple Association/Simple Association.xtuml]: strike this

File: src/org.xtuml.bp.core/models/org.xtuml.bp.core/ooaofooa/Association/Simple Association/Simple Association.xtuml [hunk 2]
```diff
@@ -370,14 +370,17 @@ INSERT INTO O_TFR
...
+for each part in parts
+  if ( not_empty part and not canParticipate)
+    select any oida related by part->R_RTO[R204]->R_OIR[R203]->O_OBJ[R201]->O_ID[R104]->O_OIDA[R105];
+    // a participant has to have an id with attributes and cardinality of one (value zero)
+    canParticipate =  not_empty oida and part.Mult == 0;
```
Comment (cortlandstarrett) [inline src/org.xtuml.bp.core/models/org.xtuml.bp.core/ooaofooa/Association/Simple Association/Simple Association.xtuml]: if ( not_empty oida and part.Mult == 0 )
  return true;
  break for;
end if;

File: src/org.xtuml.bp.core/models/org.xtuml.bp.core/ooaofooa/Association/Simple Association/Simple Association.xtuml [hunk 3]
```diff
@@ -370,14 +370,17 @@ INSERT INTO O_TFR
...
+  end if;
+end for;
+return canParticipate;',
```
Comment (cortlandstarrett) [inline src/org.xtuml.bp.core/models/org.xtuml.bp.core/ooaofooa/Association/Simple Association/Simple Association.xtuml]: return false;

**Pull request merged**